### DISCUSSION TITLE
Add createLazyResource

### DISCUSSION THREAD

**Discussion by polRk** [solidjs/solid#996] [Ideas]
I need conditional or defered resource loading.

Spec
```ts
[profile, {load, mutate, refetch}] = createLazyResource('/profile', fetcher)
```

**Comment by polRk** [comment]
In addition: add createConditionalResource

Spec
```ts
    const [ready, setReady] = createSignal(false);
    [profile, {mutate, refetch}] = createConditionalResource(ready(), '/profile', fetcher)
```

**Comment by ryansolid** [comment]
You can do both with the first argument to createResource. Returning undefined from the source defers fetching, so you can just put a conditional in there. Abstracting it into specific primitives is an option and something like solid-primitives could look at that.

Simple deferred example: https://codesandbox.io/embed/mystifying-roentgen-2o4wmxj9zy?codemirror=1
